### PR TITLE
fix: [#4204] Fix remaining eslint warnings - Rest of botbuilder-dialogs libraries - botbuilder declarative

### PR DESCRIPTION
**Pull request by erquirogasw**
Addresses # 4204
#minor

## Description
This PR fixes the ESLint and JSDoc issues in the botbuilder-dialogs-declarative library.

## Specific Changes
- Applied auto-fixes with yarn lint --fix
	- Fix spacing
	- Fix indentation
	- Add blank line in between comments
	- Add trailing commas
	- Missing parameters in descriptions
- Applied manual fixes for:
	- Missing documentation
	- Unnecessary escape characters
	- Disabling detect-non-literal-fs-filename rule to allow dynamic path for files

## Testing
_Library eslint status_
Screenshot 2022-05-04 114924

_Test:compat_
Screenshot 2022-05-04 114905

### CODE REVIEW

File: libraries/botbuilder-dialogs-declarative/src/resources/resource.ts [hunk 1]
```diff
@@ -16,13 +16,17 @@ export abstract class Resource {
...
+     * @returns The resource id.
      */
     public get id(): string {
         return this._id;
     }
 
     /**
      * The full path to the resource on disk
+     *
+     * @returns The full to the resource.
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-declarative/src/resources/resource.ts:29]:
```suggestion
     * @returns The full path to the resource.
```

File: libraries/botbuilder-dialogs-declarative/src/resources/resourceExplorer.ts [hunk 2]
```diff
@@ -43,6 +43,9 @@ export class ResourceExplorer {
      * @param {ResourceExplorerOptions} options The configuration options.
      */
     public constructor(options?: ResourceExplorerOptions);
+    /**
+     * @param providersOrOptions The list of [ResourceProvider](botbuilder-dialogs-declarative.ResourceProvider) or configuration options to initialize the current instance.
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-declarative/src/resources/resourceExplorer.ts:47]: I think this won't work to resolve the class.
```suggestion
     * @param providersOrOptions The list of [ResourceProvider](xref:botbuilder-dialogs-declarative.ResourceProvider) or configuration options to initialize the current instance.
```

**Review by ceciliaavila — APPROVED** [review]

**Review by sw-joelmut — COMMENTED** [review]
Apply the changes suggested by Ceci and add the arrow function to function in the description

**Review by sw-joelmut — APPROVED** [review]

**Review by ceciliaavila — APPROVED** [review]

**Pull request closed without merging**